vizro-core/tests/unit/vizro/models/_components/test_aggrid.py

```diff
@@ -0,0 +1,156 @@
+"""Unit tests for vizro.models.AgGrid."""
+
+import dash_ag_grid as dag
+import pytest
+from asserts import assert_component_equal
+from dash import dcc, html
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+import vizro.plotly.express as px
+from vizro.managers import data_manager
+from vizro.models._action._action import Action
+from vizro.tables import dash_ag_grid
+
+
+@pytest.fixture
+def dash_ag_grid_with_arguments():
+    return dash_ag_grid(data_frame=px.data.gapminder(), defaultColDef={"resizable": False, "sortable": False})
+
+
+@pytest.fixture
+def dash_ag_grid_with_str_dataframe():
+    return dash_ag_grid(data_frame="gapminder")
+
+
+class TestDunderMethodsAgGrid:
+    def test_create_graph_mandatory_only(self, standard_ag_grid):
+        ag_grid = vm.AgGrid(figure=standard_ag_grid)
+
+        assert hasattr(ag_grid, "id")
+        assert ag_grid.type == "ag_grid"
+        assert ag_grid.figure == standard_ag_grid
+        assert ag_grid.actions == []
+
+    @pytest.mark.parametrize("id", ["id_1", "id_2"])
```

Comment:
Why is this parameterised? The two cases are functionality identically aren't they?

Reply:
This was done the same on the table - is suppose it is to check that there is no hidden ID, and that it gets dynamically set, but it's a bit silly indeed